docs/source/getting-started/tabular-data/keys.ipynb

```diff
@@ -0,0 +1,280 @@
+      "text/plain": [
+       "<Figure size 1000x600 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "sum_year_actual = (df.group_by(\"YEAR\").agg(pl.col(\"HWUSUAL\").sum().alias(\"Actual Sum\")).sort(\"YEAR\")).collect()\n",
+    "\n",
+    "sum_year_dp = (\n",
+    "    context.query().group_by(\"YEAR\").agg(\n",
+    "        pl.col(\"HWUSUAL\").fill_null(40.).dp.sum((0,100)).alias(\"DP Sum\")\n",
+    "    ).sort(\"YEAR\")).release().collect()\n",
+    "\n",
+    "plt.figure(figsize=(10, 6))\n",
+    "sns.lineplot(x=sum_year_actual[\"YEAR\"].to_list(), y=sum_year_actual[\"Actual Sum\"].to_list(), marker=\"o\", label=\"Actual\")\n",
+    "sns.lineplot(x=sum_year_dp[\"YEAR\"].to_list(), y=sum_year_dp[\"DP Sum\"].to_list(), marker=\"o\", label=\"DP: Private\")\n",
+    "plt.title('Actual vs DP Sums by Year')\n",
+    "plt.xlabel('Year')\n",
+    "plt.ylabel('Sum')\n",
+    "plt.legend(title='Legend')\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Grouping By Public Lengths\n",
+    "\n",
+    "Queries whose functions don't depend on the partition length may not have a significantly different utility if the `public_info` parameter is public or private. \n",
+    "\n",
+    "One example scenario is the sum of total hours worked per year where the data is being aggregated over a specific key and the noise addition is independent of the partition lengths being revealed. \n",
+    "\n",
+    "However, be cautious when you allow the parameter to be public and enable it to be public only in settings where you are confident the partition length will not reveal any sensitive information. \n"
```

Comment:
I would draw an equivalence in this section to bounded-DP in each partition. I think it would help ground the concepts in terminology readers might already be familiar with.

Reply:
could you explain this more?